source/FAST/Algorithms/ImagePatch/ImageToBatchGenerator.cpp

```diff
@@ -61,13 +68,13 @@ void ImageToBatchGenerator::generateStream() {
 }
 
 void ImageToBatchGenerator::execute() {
-    if(m_maxBatchSize == 1)
+    if(m_maxBatchSize == -1)
         throw Exception("Max batch size must be given to the ImageToBatchGenerator");
 
     if(!m_streamIsStarted) {
         m_streamIsStarted = true;
         mParent = mInputConnections[0];
-        mInputConnections.clear(); // Severe the connection
+        //mInputConnections.clear(); // Severe the connection  @FIXME: haivng this results in "Input port X is missing its required connection when streaming. Hence, I removed this. Why was it used?"
```

Comment:
In which case did you get this error?
The ImageToBatchGenerator unit test is working fine with this.